Cast error building with g++

Hi, I am getting these errors when compiling with g++. Gambit tests are ok so assuming it is a gerbil issue? Dont really understand how to locate the error as it is in the .c file.

```
... compile os/signal
... compile os/signal-handler
In file included from signal-handler__0.c:271:
signal-handler__0.c: In function 'long int ___H_signal_2d_handler____0(___processor_state)':
/usr/local/gambit/v4.9.0/include/gambit.h:1707:27: error: invalid conversion from 'void*' to 'void (*)(int)' [-fpermissive]
 #define ___CAST(type,val)((type)(val))
                          ~^~~~~~~~~~~~
/usr/local/gambit/v4.9.0/include/gambit.h:7465:42: note: in definition of macro '___CFUN_ASSIGN_POINTER'
 #define ___CFUN_ASSIGN_POINTER(r,val)r = val;
                                          ^~~
/usr/local/gambit/v4.9.0/include/gambit.h:7448:31: note: in expansion of macro '___CAST'
 #define ___CFUN_CAST(type,arg)___CAST(type,arg)
                               RROR IN ##main -- target compilation or link failed while compiling "/Users/izzy/local/gerbil/lib/std/os/signal-handler__0.scm"
*** ERROR IN gxc#compile-scm-file__% -- Syntax Error
*** ERROR IN ?
--- Syntax Error at compile: Compilation error; gsc exit with nonzero status
... form:   "/Users/izzy/local/gerbil/lib/std/os/signal-handler__0.scm"
... detail: 17920
*** ERROR; build failed

```

```
In file included from signal-handler__0.c:271:
signal-handler__0.c: In function 'long int ___H_signal_2d_handler____0(___processor_state)':
signal-handler__0.c:6710:17: error: invalid conversion from 'void (*)(int)' to 'void*' [-fpermissive]
 ___return(signal(___arg1, (void (*)(int))(___arg2)));
           ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/gambit/v4.9.0/include/gambit.h:7465:42: note: in definition of macro '___CFUN_ASSIGN_POINTER'
 #define ___CFUN_ASSIGN_POINTER(r,val)r = val;
                                          ^~~
signal-handler__0.c:6710:1: note: in expansion of macro '___return'
 ___return(signal(___arg1, (void (*)(int))(___arg2)));
 ^~~~~~~~~
*** ERROR IN ##main -- target compilation or link failed while compiling "/Users/izzy/local/gerbil/lib/std/os/signal-handler__0.scm"
*** ERROR IN gxc#compile-scm-file__% -- Syntax Error
*** ERROR IN ?
--- Syntax Error at compile: Compilation error; gsc exit with nonzero status
... form:   "/Users/izzy/local/gerbil/lib/std/os/signal-handler__0.scm"
... detail: 17920
*** ERROR; build failed
```

--- a/src/std/os/signal-handler.ss
+++ b/src/std/os/signal-handler.ss
@@ -174,12 +174,13 @@ package: std/os
      (namespace ("std/os/signal-handler#"
                  SIG_IGN
                  set-signal!))
-     
+
      (define SIG_IGN
        ((c-lambda () (pointer "void") "___return((void *)SIG_IGN);")))
 
      (define set-signal!
-       (c-lambda (int (pointer "void")) (pointer "void") "signal"))))
+       (c-lambda (int (pointer "void")) (pointer "void")
+            "___return((void*)signal(___arg1, (void (*)(int))(___arg2)));"))))
 
   (else
    (def (make-signal-handler)

